Remotion/Security/Core/Metadata/EnumerationReflector.cs

```diff
@@ -52,7 +52,8 @@ public EnumerationReflector ()
       Dictionary<Enum, EnumValueInfo> enumValueInfos = new Dictionary<Enum, EnumValueInfo> ();
       for (int i = 0; i < values.Count; i++)
       {
-        Enum value = (Enum) values[i];
+        // TODO: Add notnull assertion
+        Enum value = (Enum) values[i]!;
```

Comment:
Could you check, please, I think we can change `IList` in line 50 to `Enum[]` . That way, we don't need a not-null assertion or a bang. Not sure if doing this change belongs in the current PR or should be a separate task (read, we could just fix it if you like).

Reply:
Implicit conversion does not work at compile type, explicit cast throws at runtime "Unable to cast object of type '<enumtype>[]' to type 'System.Enum[]'".

The following would work:
```csharp
var values = Enum.GetValues (type).Cast<Enum>().ToArray();

Dictionary<Enum, EnumValueInfo> enumValueInfos = new Dictionary<Enum, EnumValueInfo> ();
for (int i = 0; i < values.Length; i++)
{
  // TODO: Add notnull assertion
  Enum value = (Enum) values[i]!;
  enumValueInfos.Add (value, GetValue (value, cache));
}
```

Reply:
Hmm... yeah, don't think this is worth it (extra runtime cost and code busy work to get pseudo-typesafety). probably should just stick with the bang and be done with it.

Reply:
Alright, then I'll just remove the TODO and leave the bang.